src/client.rs

```diff
@@ -966,28 +961,62 @@ impl Client {
         ))
     }
 
+    /// Mounts a blob to the provided reference, from the given source
+    pub async fn mount_blob(
+        &self,
+        image: &Reference,
+        source: &Reference,
+        digest: &str,
+    ) -> Result<()> {
+        let base_url = self.to_v2_blob_upload_url(image);
+        let url = format!("{}?mount={}&from={}", base_url, digest, source.repository());
```

Comment:
I might be a bit paranoid, but I would prefer to have the url built using the `Url` crate.

Using the [`query_pairs_mut`](https://docs.rs/url/2.3.1/url/struct.Url.html#method.query_pairs_mut) method would allow us to do proper encoding of the query parameters